### PR TITLE
Validate canvas configuration in configure()

### PR DESCRIPTION
**Pull request by kainino0x**
~**Dependent on #2823**~

Moves the validation of the GPUCanvasConfiguration to configure(), instead of doing it implicitly later in getCurrentTexture().

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on May 16, 2022, 10:29 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [CSS Spec Preprocessor](https://api.csswg.org/bikeshed/) - CSS Spec Preprocessor is the web service used to build Bikeshed specs.

:link: [Related URL](https://api.csswg.org/bikeshed/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fkainino0x%2Fgpuweb%2F25fa086307e93dffc9cf1b1b34caeb7ce4e48c60%2Fspec%2Findex.bs&force=1&md-warning=not%20ready)

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>500 Internal Server Error</title>
</head><body>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error or
misconfiguration and was unable to complete
your request.</p>
<p>Please contact the server administrator at 
 [no address given] to inform them of the time this error occurred,
 and the actions you performed just before this error.</p>
<p>More information about this error may be available
in the server error log.</p>
<hr>
<address>Apache/2.4.10 (Debian) Server at api.csswg.org Port 443</address>
</body></html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20gpuweb/gpuweb%232821.)._
</details>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -9757,11 +9742,27 @@ interface GPUCanvasContext {
...
+                    Implementations may deduplicate this validation, but **must** still resurface the validation error here.
+
+                    Note: If a previously presented texture from |this| matches the required criteria,
+                    its GPU memory may be re-used. This optimization is unobservable; the returned texture is always cleared.
+
+                    Issue: FIXME: If this fails, can you attempt to getCurrentTexture again (e.g. after freeing some resources)?
+                    If yes, how does getCurrentTexture know to try again instead of returning the same failed texture?
+                    It's not possible to check `valid` from the content timeline.
+                    There's also a related issue here that `destroyed` is checked from the content timeline (technically possible
+                    but not intended).
```
Comment (kainino0x) [inline spec/index.bs:9765]: note to self: I need to figure this out before unmarking as draft

Comment (kainino0x) [reply spec/index.bs:9765]: never mind, it doesn't matter because you can't reliably know that it failed in the same frame anyway.

**Comment by kainino0x** [comment]
@toji PTAL (ignore the first two commits, those are #2823)

**Review by toji — APPROVED** [review]
LGTM. There's some minor questions that I have about how this will interact with the canvas size changes we discussed this morning, but I'm happy to leave those to the follow-up change. This feels significantly cleaner than what we previously had in the meantime.

**Pull request merged**